lib/ecto/changeset.ex

```diff
@@ -287,6 +287,8 @@ defmodule Ecto.Changeset do
   The function is meant for working with data internal to the application.
   Because of that neither validation nor casting is performed. This means
   `change/2` expects the keys in the `changes` map or keyword to be atoms.
+  If any value in the given `changes` does not match the
+  type for that key in the schema (or `types` map), an error is raised.
```

Comment:
I'm not sure either. FWIW we don't perform casting in the sense that we don't use the return value of `cast`, we just use cast to check types. but maybe treating it as essentially `cast!` is still considered casting.